### PR TITLE
feat(connector): [Bitpay] Add order id as the reference id

### LINKED ISSUE #2325: [FEATURE]: [Bitpay] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by ashh1215**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

[Relevant documentation](https://developer.bitpay.com/docs/create-invoice#notificationurl)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes #2325

## How did you test it?

Make any Payment for connector Bitpay and see that you are getting "reference_id" field in the payments response of hyperswitch. It should not be null.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by ashh1215** [comment]
@SanchithHegde Please do review and suggest changes if required

**Review by deepanshu-iiitu — APPROVED** [review]

File: crates/router/src/connector/bitpay/transformers.rs [hunk 1]
```diff
@@ -128,7 +128,7 @@ impl<F, T>
             .data
             .url
             .map(|x| services::RedirectForm::from((x, services::Method::Get)));
-        let connector_id = types::ResponseId::ConnectorTransactionId(item.response.data.id);
+        let connector_id = types::ResponseId::ConnectorTransactionId(item.response.data.id.clone());
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bitpay/transformers.rs:131]: Add a field `order_id` in `BitpayPaymentResponseData` and use it in connector_response_reference_id

File: crates/router/src/connector/bitpay/transformers.rs [hunk 2]
```diff
@@ -138,7 +138,7 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(item.response.data.id),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bitpay/transformers.rs:141]:
```suggestion
                connector_response_reference_id: item.response.data.order_id.or(Some(connector_id)),
```

Comment (deepanshu-iiitu) [reply crates/router/src/connector/bitpay/transformers.rs:141]: @ashh1215

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Comment by AkshayaFoiger** [comment]
Hey @ashh1215 
Thanks for your interest in contributing to hyperswitch.
Let us know if you need any assistance from our end.
.
Also, even if hacktoberfest is over, we should celebrate open source everyday and we are open for more contributions from you.
We would still be rewarding folks with goodies even if the PR gets merged post hacktoberfest.
May the Source be with you!

**Comment by ashh1215** [comment]
@deepanshu-iiitu 
I've made the changes. Please review.
Thank you.

**Comment by AkshayaFoiger** [comment]
Hey @deepanshu-iiitu  ,
Pls address the CI fails
Run `cargo clippy` and addressed lints thrown by it.

**Review by deepanshu-iiitu — APPROVED** [review]

**Comment by AkshayaFoiger** [comment]
@ashh1215, please run `cargo +nightly fmt`

**Comment by ashh1215** [comment]
@deepanshu-iiitu I've fixed the issues. It should work now.

**Review by deepanshu-iiitu — APPROVED** [review]
LGTM

**Review by AkshayaFoiger — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @ashh1215!

**Comment by deepanshu-iiitu** [comment]
Hey @ashh1215  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**